Israel-Laguan/demo-marketplace
# Integrate Prisma, Docker Compose, PostgreSQL, and Next.js Models

# Description

This PR brings a robust database setup and integration to our project, including:

- Prisma: ORM for effortless database interactions
- Docker Compose: Streamlined PostgreSQL setup and management
- Database Models: Clearly defined data structures
- Query functions: CRUD functionalities for database access
- Development Environment Enhancements: VS Code extension for PostgreSQL

## Key Features

- Dockerized PostgreSQL: Spinning up a database is a breeze with Docker Compose.
- Prisma Efficiency: Interact with the database using a user-friendly ORM.
- TypeScript for Clarity: Type safety and maintainability throughout the codebase.
- API Routes for Data Access: Structured endpoints for seamless data management.
- Developer-Friendly Setup: VS Code extension streamlines PostgreSQL interactions.

## How to test

1. Install Prisma CLI globally: `npm install -g prisma`
2. `npx prisma migrate dev`

![image](https://github.com/Israel-Laguan/demo-marketplace/assets/36519478/789ab016-5d21-48f2-bad7-fcfe00ae0363)

3. `npx prisma db seed`

![image](https://github.com/Israel-Laguan/demo-marketplace/assets/36519478/17e6d898-b733-404e-9de7-d566a5c7478c)

4. Run Prisma locally: `npx prisma studio` (accesses the Prisma Studio UI)

At the end you should see the products on your DB

![image](https://github.com/Israel-Laguan/demo-marketplace/assets/36519478/6be25d01-0c58-421d-8bbf-8a6a9a16d865)

![image](https://github.com/Israel-Laguan/demo-marketplace/assets/36519478/aff10d38-e3ce-4705-bf84-407827866531)

![image](https://github.com/Israel-Laguan/demo-marketplace/assets/36519478/9c7e5012-8b13-42a5-9101-422785062d62)


### My Development Environment:

Developed on Debian 12
Utilized Docker Desktop for containerization

I'm eager to receive your feedback and merge this PR to enhance our project's data capabilities!
